### PR TITLE
refresh the ebs-csi-driver to latest CSI driver version 1.31.0

### PR DESCRIPTION
**Pull request by gauravAggrahari**

### CODE REVIEW

File: config-templates/aws/aws-ebs-csi-driver/changelog.json [hunk 1]
```diff
@@ -1,4 +1,20 @@
 [
+    {
+        "1.12.0": {
```
Comment (nkkashyap) [inline config-templates/aws/aws-ebs-csi-driver/changelog.json]: it should be `1.31.0`

**Review by nkkashyap — CHANGES_REQUESTED** [review]

File: config-templates/aws/aws-ebs-csi-driver/1.31.0/metadata.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      "name": "provider-logo",
+      "value": "logo/aws-ebs-csi-driver.svg"
+    },
+    {
+      "name": "supported-envs",
+      "value": "AWS"
+    },
+    {
+      "name": "supported-ocp-versions",
+      "value": "4.14,4.15"
```
Comment (mssachan) [inline config-templates/aws/aws-ebs-csi-driver/1.31.0/metadata.json]: Add space after comma

File: config-templates/aws/aws-ebs-csi-driver/1.31.0/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,712 @@
...
+      labels:
+        app.kubernetes.io/name: aws-ebs-csi-driver
+        app.kubernetes.io/version: 1.31.0
+    spec:
+      selector:
+        matchLabels:
+          app: ebs-csi-controller
+          app.kubernetes.io/name: aws-ebs-csi-driver
+          app.kubernetes.io/version: 1.31.0
+      minAvailable: 1
```
Comment (mssachan) [inline config-templates/aws/aws-ebs-csi-driver/1.31.0/deployment.yaml]: change it to `2`

File: config-templates/aws/aws-ebs-csi-driver/changelog.json [hunk 4]
```diff
@@ -1,4 +1,20 @@
 [
+    {
+        "1.31.0": {
+            "status": "supported",
+            "latest-revision": {
+                "revision": "1",
+                "release-date": "3 June 2024",
```
Comment (mssachan) [inline config-templates/aws/aws-ebs-csi-driver/changelog.json]: `3 June 2024` ---> `05 June 2024`

**Review by mssachan — APPROVED** [review]
Added minor comments

**Review by mssachan — APPROVED** [review]
lgtm

**Review by nkkashyap — APPROVED** [review]
lgtm

**Pull request merged**